### ISSUE TITLE
Path A: Replace hurdle mask with Tobit censored-normal likelihood

### ISSUE THREAD

**Issue by Polichinel** [views-platform/views-hydranet#36]
## Sequence Position

> **This is Step 1 of 6** in the zero-inflation instability remediation sequence: **A → E → D/B → C/F**
>
> This issue is the **recommended first step** and has no dependencies. It directly fixes the diagnosed root cause of sweep instability (gradient starvation from the hard hurdle mask). All subsequent paths depend on or build upon the outcome of this work.

---

### The diagnosed failure

The sweep isolation experiment (see `reports/sweep_isolation_plan.md`) identified `hurdle_threshold=0.0` as the primary cause of autoregressive divergence during 36-step inference. The mechanism:

1. **Gradient starvation:** The hurdle mask in `training_engine.py:172-178` zeros out regression loss for all cells where y=0. Since ~95% of PRIO-Grid cells have zero fatalities in any given month, the regression head receives gradient from only ~5% of the grid.
2. **Training non-convergence:** Regression loss oscillates wildly (2300–4300) instead of the baseline's monotonic descent (1200→221). The model never learns to predict near-zero for quiet regions.
3. **Autoregressive amplification:** During inference (`hydranet_inference.py:289`: `t0_autoreg = t1_pred.detach()`), unconstrained quiet-cell predictions feed back as input at the next step. Small positive values compound over 36 steps.
4. **Silent corruption:** `np.expm1()` on large positive values produces infinity. No post-inverse-transform integrity check exists.

### Why the baseline works

The baseline uses shrinkage loss on ALL cells. Zero cells receive a loss signal that pushes predictions toward zero. During autoregression, these near-zero predictions feed back as near-zero inputs — stable.

### Why Tobit is the correct fix

The model's regression output already applies `F.relu()` (`HydraBNrecurrentUnet_06_LSTM4.py:432,464,496`), producing non-negative values in log1p-space. As Zhang et al. (2021) demonstrate, `max(0, z*)` is *literally* the Type-I Tobit observation equation — the model architecturally implements left-censoring at zero. We are already running a censored regression architecture. We're just using the wrong loss function for it.

The Tobit likelihood treats y=0 not as "ignore this cell" (hurdle) but as "the latent intensity z* was ≤ 0, so the observed value was censored to 0." This provides dense gradient from ALL cells:

```
L(y, μ, σ) = {
    -log Φ(-μ/σ)                          if y = 0  (censored: latent z* ≤ 0)
    -log φ((y - μ)/σ) + log σ             if y > 0  (observed: standard regression)
}
```

where Φ is the standard normal CDF and φ is the standard normal PDF.

For y=0 cells, the gradient of `-log Φ(-μ/σ)` pushes μ toward negative values. After ReLU, the output is clamped to 0. But the gradient flows through the pre-ReLU activations, teaching the model "this cell should be quiet." During autoregression, these cells output 0 and feed back 0 — exactly the stable behavior of the baseline, but achieved through a principled statistical mechanism rather than blanket regression over all cells.

---

## Literature Support

Five independent research groups have validated deep Tobit in different settings:

| Paper | Authors | Year | Key contribution | Link |
|-------|---------|------|-----------------|------|
| Deep Censored Regression | Dănăilă & Buiu | 2024 | Tobit loss for DNNs. Three-component loss L = L_p + L_l + L_u. Reparametrized variant (γ=1/σ, δ(x)=γ·f(x)) has **globally concave** log-likelihood — no local optima. | [Springer](https://link.springer.com/article/10.1007/s10044-024-01216-9) |
| Penalized Tobit | Jacobson & Zou | 2022/2023 | Olsen reparameterization (δ=β/σ, γ²=1/σ²) makes Tobit NLL convex. MSE advantage over OLS grows with censoring proportion q — at q=0.5, Tobit has 2–5× lower MSE. **At our q≈0.95, the advantage is massive.** Strong oracle property with SCAD penalty. | [arXiv](https://arxiv.org/abs/2203.02601), [JBES](https://ideas.repec.org/a/taf/jnlbes/v42y2024i1p286-297.html) |
| Deep Tobit Networks | Zhang, Li, Song, Ning | 2021 | DTN-I: ReLU output ≡ Type-I Tobit censoring. DTN-II: Heckman selection model with coupled selection + outcome equations. Outperforms standard DNNs on censored microeconometric data. | [ScienceDirect](https://www.sciencedirect.com/science/article/abs/pii/S0893608021003531) |
| Deep Tobit Model | Wu, Hu, Ye, Chen | 2026 | Two-stage variable selection with convergence rate and selection consistency guarantees. Confirms framework maturity. | [Springer](https://link.springer.com/article/10.1007/s10985-026-09690-5) |
| TOBART | O'Neill | 2024 | Tobit censoring + Bayesian Additive Regression Trees. Nonparametric variant with Dirichlet process mixture errors relaxes normality. Confirms Tobit works with nonlinear function approximators. | [arXiv](https://arxiv.org/abs/2211.07506) |

---

### 1. New loss class: `TobitLoss`

Create `views_hydranet/utils/tobit_loss.py`:

```python
class TobitLoss(nn.Module):
    def __init__(self, sigma=1.0):
        super().__init__()
        self.sigma = sigma

    def forward(self, pred, target):
        # pred is PRE-ReLU (latent μ), target is in log1p-space
        z = pred / self.sigma
        censored = target == 0
        # Censored cells: -log Φ(-z)
        loss_censored = -torch.distributions.Normal(0, 1).log_cdf(-z[censored])
        # Observed cells: Gaussian NLL
        loss_observed = 0.5 * ((target[~censored] - pred[~censored]) / self.sigma) ** 2 + math.log(self.sigma)
        return loss_censored.mean() + loss_observed.mean()
```

### 2. Expose pre-ReLU activations

The Tobit loss needs the latent μ (pre-ReLU), not the post-ReLU output. The current architecture applies ReLU inside the decoder heads (`HydraBNrecurrentUnet_06_LSTM4.py:432,464,496`).

**Options (in order of preference):**
- **(a)** Modify `ModelOutput` to include `reg_latent` (pre-ReLU) alongside `reg` (post-ReLU). Training uses `reg_latent` for the loss; inference uses `reg` for autoregressive feedback.
- **(b)** Move ReLU out of the model and into the training/inference code. Model returns raw latent; ReLU applied as needed.
- **(c)** Register a forward hook to capture pre-activation values.

Option (a) is cleanest — the model returns both representations, and consumers choose which they need.

### 3. Register in loss registry

Add `'tobit'` to `LOSS_REG_REGISTRY` in `views_hydranet/utils/utils.py:119`.

### 4. Config parameters

```python
'loss_reg': 'tobit',
'loss_reg_sigma': 1.0,  # fixed scale parameter
```

### 5. σ handling variants (incremental)

| Variant | σ treatment | When to try |
|---------|------------|-------------|
| Fixed σ=1.0 | Hyperparameter | **Start here** |
| Learned σ | Single `nn.Parameter` | If fixed σ is sensitive to value choice |
| Reparametrized γ=1/σ | Train γ, compute σ=1/γ | If learned σ causes instability (Dănăilă & Buiu 2024) |
| Heteroscedastic σ(x) | Network predicts per-cell σ | Future refinement only — highest risk |

---

### Isolation test

1. Create `config_hp_s2a_tobit.py` in `reports/configs_isolation/`
2. Identical to S0 baseline but with `'loss_reg': 'tobit'`, `'loss_reg_sigma': 1.0`
3. Run single calibration (`n_posterior_samples=3`)
4. **Pass criteria** (from `reports/s0_baseline_metrics.md`):
   - Evaluation completes without crash or infinity
   - CRPS values within ~2× of baseline (no metric exceeds ~0.6)
   - MCR values within ~3× of baseline (no MCR exceeds ~35)
   - Brier scores within ~2× of baseline (no Brier exceeds ~0.02)

### Training dynamics

- Regression loss should show monotonic descent (like S0/S1), not oscillation (like S2-hurdle)
- Training should converge within the same number of lessons as baseline

### Regression suite

```bash
conda run -n views-hydranet-env pytest tests/ -v --ignore=tests/test_eval_integration_toy.py
conda run -n views-hydranet-env ruff check .
```

---

## Acceptance Criteria

- [ ] `TobitLoss` class implemented with fixed-σ variant
- [ ] Pre-ReLU latent values accessible for loss computation
- [ ] Registered in `LOSS_REG_REGISTRY` as `'tobit'`
- [ ] Config schema updated in `config_initializer.py` to accept `loss_reg: 'tobit'`
- [ ] Isolation test (S2a-Tobit) passes with finite metrics in baseline ballpark
- [ ] Training dynamics show monotonic convergence, not oscillation
- [ ] All existing tests pass (no regression)
- [ ] ADR documenting the Tobit design rationale and σ-handling decision

---

## What This Does NOT Solve

- **Heavy-tailed extremes:** Tobit assumes normal latent distribution. Conflict fatalities have power-law tails. See Path D (tail-aware extension).
- **Autoregressive exposure bias:** Even with perfect zero-cell predictions, 36-step AR compounds errors from teacher-forcing mismatch. See Path E (GTF/scheduled sampling).
- **Selection bias in active cells:** The regression head is trained on a censored sample (only y>0 contributes to the Gaussian NLL component). See Path B (Heckman correction).

These are addressed by subsequent issues in the sequence.

**Comment by Polichinel** [comment]
## Path A — Experimentally Validated (Gate 1 PASS)

### S2a-150 results (2026-05-29)

Tobit censored-normal with `sigma=1.0`, 150 lessons. Training converges by ~lesson 60 (plateau).

**Step-wise CRPS vs S0 baseline:**
| Target | S0 | S2a-150 | Change |
|--------|-----|---------|--------|
| lr_sb | 0.291 | 0.166 | **-43%** |
| lr_ns | 0.104 | 0.047 | **-55%** |
| lr_os | 0.255 | 0.052 | **-80%** |

**MCR recovery (target ~1.0):**
| Target | S2a (20 lessons) | S2a (150 lessons) |
|--------|-----------------|-------------------|
| lr_sb | 0.040 | **0.558** |
| lr_ns | 0.003 | **0.811** |
| lr_os | 0.025 | 0.015 (stuck) |

Key findings:
- Gate 1 **PASSES** — Tobit restores convergence and improves CRPS 43-80%
- MCR recovers with more training (magnitude calibration improves from ~0.04 to ~0.6-0.8 for sb/ns)
- lr_os MCR stuck near 0 — needs dedicated attention (per-target sigma or Path D)
- Optimal training: ~60-80 lessons for Tobit (loss plateaus by 60)
- Exposure bias quantified: step-wise MCR (0.56) vs month-wise MCR (0.98) gap → Path E baseline

Proceed to Path E (#37).

**Comment by Polichinel** [comment]
Closed — Path A implemented in commit 56194d2, validated by S2a sweep (Gate 1 PASS). See issue #42 for full roadmap status.
